sktime/annotation/e_agglo.py

```diff
@@ -0,0 +1,435 @@
+        self :
+            Reference to self.
+        """
+        self._X = X
+
+        assert self.alpha > 0 and self.alpha <= 2
+
+        self._process_data(X)
+
+        # find which clusters optimize the GOF and then update the distances
+        for K in range(self.n_cluster - 1, 2 * self.n_cluster - 2):
+            i, j = self._find_closest(K)
+            self._update_distances(i, j, K)
+
+        def filter_na(i):
+            return list(
+                filter(
+                    lambda v: v == v,
+                    self.progression[
+                        i,
+                    ],
+                )
+            )
+
+        # penalize the GOF statistic
+        if self.penalty is not None:
+            penalty_func = get_penalty_func(self.penalty)
+            cps = [filter_na(i) for i in range(len(self.progression))]
+            self.gof += list(map(penalty_func, cps))
+
+        # get the set of change points for the "best" clustering
+        idx = np.argmax(self.gof)
+        self._estimates = np.sort(filter_na(idx))
+
+        # remove change point N+1 if a cyclic merger was performed
+        self._estimates = (
+            self._estimates[:-1] if self._estimates[0] != 0 else self._estimates
+        )
+
+        # create final membership vector
+        def get_cluster(estimates):
+            return np.repeat(
+                range(len(np.diff(estimates))), np.diff(estimates).astype(int)
+            )
+
+        if self._estimates[0] == 0:
+            self.cluster = get_cluster(self._estimates)
+        else:
+            tmp = get_cluster(np.append([0], self._estimates))
+            self.cluster = np.append(tmp, np.repeat(0, X.shape[0] - len(tmp)))
+
+        return self
+
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        private _transform containing core logic, called from transform
+
+        Parameters
+        ----------
+        X : Series of mtype X_inner_mtype
+            Data to be transformed
+        y : Series of mtype y_inner_mtype, default=None
+            Not required for this unsupervised transform.
+
+        Returns
+        -------
+        cluster
+            numeric representation of cluster membership for each row of X.
+        """
+        # fit again if indices not seen, but don't store anything
+        if not X.index.equals(self._X.index):
+            X_full = X.combine_first(self._X)
+            new_eagglo = EAGGLO(
+                member=self.member,
+                alpha=self.alpha,
+                penalty=self.penalty,
+            ).fit(X_full)
+            warnings.warn(
+                "Warning: Input data X differs from that given to fit(). "
+                "Refitting with both the data in fit and new input data, not storing "
+                "updated public class attributes. For this, explicitly use fit(X) or "
+                "fit_transform(X)."
+            )
+            return new_eagglo.cluster
```

Comment:
This is quite unexpected for me and suggests that there might be an ill fit for using a transformer here. Feel free to ignore this comment since I don't have a suggestion to improve this atm :slightly_smiling_face: